src/command.rs

```diff
@@ -139,3 +181,65 @@ impl ShellCommand for ExternalCommand {
         Ok(child.stdout.take())
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct LlmCommand {
+    prompt: String,
+    openai_client: OpenAIClient,
+}
+
+impl LlmCommand {
+    pub fn new(prompt: String, openai_client: OpenAIClient) -> Self {
+        LlmCommand {
+            prompt,
+            openai_client,
+        }
+    }
+
+    pub async fn generate_response(&self, input: Option<String>) -> Result<String, Box<dyn Error>> {
+        debug!("Received input: {:?}", input);
+        let context = if let Some(input) = input {
+            format!("{}: {}", self.prompt, input)
```

Comment:
I still think the prompt to the llm should be more structured, especially if it receives input. I'd imagine that will get the LLM to perform better.

I don't know if you were going to do it in another issue or not. If you are, could you create an issue and link it.

Reply:
We'll do this in another PR

Reply:
#76